rustls/src/msgs/handshake.rs

```diff
@@ -1029,9 +1037,11 @@ impl ClientHelloPayload {
     }
 
     pub fn set_psk_binder(&mut self, binder: impl Into<Vec<u8>>) {
-        let last_extension = self.extensions.last_mut();
-        if let Some(ClientExtension::PresharedKey(ref mut offer)) = last_extension {
-            offer.binders[0] = PresharedKeyBinder::new(binder.into());
+        if let Some(ref mut extensions) = &mut self.extensions {
+            let last_extension = extensions.last_mut();
```

Comment:
Notice that the `Option` doesn't help this function; it makes it more complicated.